controller/array_action/array_mediator_interface.py

```diff
@@ -54,27 +54,26 @@ def create_volume(self, volume_name, size_in_bytes, space_efficiency, pool):
         raise NotImplementedError
 
     @abstractmethod
-    def copy_to_existing_volume_from_source(self, name, source_name, source_capacity_in_bytes,
-                                            minimum_volume_size_in_bytes, pool=None):
+    def copy_to_existing_volume_from_source(self, volume_id, source_id, source_capacity_in_bytes,
+                                            minimum_volume_size_in_bytes):
         """
         This function should create a volume from source volume or snapshot in the storage system.
 
         Args:
-            name                            : name of the volume to be created in the storage system
-            source_name                     : name of source to create from
+            volume_id                       : id of the volume to be created in the storage system
+            source_id                       : id of source to create from
             source_capacity_in_bytes        : capacity of source to create from
             minimum_volume_size_in_bytes    : if source capacity is lower than this value volume will
                                             be increased to this value
-            pool                            : pool of the volume and source object to find them more efficiently.
 
         Returns:
             Volume
 
         Raises:
             ObjectNotFoundError
             IllegalObjectName
```

Comment:
which object name?

Reply:
the [same one](https://github.com/IBM/ibm-block-csi-driver/blob/cf76881f4aa478cd1fc6676f9b70363952cee78a/controller/array_action/array_mediator_svc.py#L230) there was before.. (but now we are getting it in the mediator, the rest of the calls didn't change so the possible errors returned stayed the same)

Reply:
since the server no longer sends a name, I don't think it should except/expect this error, it sounds to me like an internal mediator implementation detail (not relevant to e.g. xiv).

and even if for some reason it gets raised, it should not lead to `INVALID_ARGUMENT`, since at this point we know the request values (specifically the source **id**) are valid

Reply:
then its a bug, cause nothing actually changed

Reply:
I agree that according to my suggested interface, some mediators will be "supposed" to internally except `IllegalObjectName` accordingly, but I'm OK if they will "blindly" rely on the name to be valid, since they now get it from **the storage**.

Reply:
ok, done